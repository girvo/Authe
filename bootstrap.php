<?php
/**
 * Bootstrap file for bringing in autloading, etc
 */
use Slim\Slim;

require("vendor/autoload.php");

$app = new Slim();

$app->get("/", function() {
    print("<h1>Hello, World!</h1>");
});

$app->run();