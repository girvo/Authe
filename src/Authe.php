<?php

namespace Authe;

class Authe
{
    private $TokenHandler;
    private $DBHandler;
    
    public function __construct($TokenHandler = null, $DBHandler = null) {
        if ($TokenHandler)
            $this->TokenHandler = $TokenHandler;
        if ($DBHandler)
            $this->DBHanlder = $DBHandler;
    }
    
    /**
     * A wrapper function to be pulled apart later
     */
    public function loginAndGetToken($username, $password) {
        
        return FALSE;
    }
}