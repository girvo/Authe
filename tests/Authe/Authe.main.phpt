<?php

use Tester\Assert;
use Authe\Authe;

// Load the bootstrap
require(__DIR__ . "/../bootstrap.php");

function testCheckAndDoLogin() {
    $Authe = new Authe();

    $username = "testusername";
    $password = "testpassword";
    
    $loginResult = $Authe->checkAndDoLogin($username, $password);
    
    Assert::same($loginResult, "test");
}


// now run the tests
testCheckAndDoLogin();