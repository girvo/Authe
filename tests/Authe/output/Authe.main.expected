array('success' => FALSE, 'data' => array())